src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -529,7 +532,7 @@ QgsSpatiaLiteProvider::QgsSpatiaLiteProvider( QString const &uri )
     return;
   }
   mEnabledCapabilities = mPrimaryKey.isEmpty() ? QgsVectorDataProvider::Capabilities() : ( QgsVectorDataProvider::SelectAtId );
-  if ( ( mTableBased || mViewBased ) &&  !mReadOnly )
+  if ( mTableBased )
```

Comment:
Should that be `if ( mTableBased && !mReadOnly )`? Readonly table based connections have previously not been handled in this block and now are